net/base/directory_lister.h

```diff
@@ -21,11 +21,119 @@ class TaskRunner;
 
 namespace net {
 
-// This class provides an API for asynchronously listing the contents of a
-// directory on the filesystem.  It runs a task on a background thread, and
-// enumerates all files in the specified directory on that thread.  Destroying
-// the lister cancels the list operation.  The DirectoryLister must only be
-// used on a thread with a MessageLoop.
+#if defined(STARBOARD)
+// TODO b/296715826 Fix stubbed out class.
+class NET_EXPORT DirectoryLister  {
+ public:
+  // Represents one file found.
+  struct DirectoryListerData {
+    base::FileEnumerator::FileInfo info;
+    base::FilePath path;
+    base::FilePath absolute_path;
+  };
+
+  // Implement this class to receive directory entries.
+  class DirectoryListerDelegate {
```

Comment:
You should be able to turn this into a full stub, e.g. `class DirectoryListerDelegate {}` that does nothing. Unless it's actually referenced from somewhere.

Reply:
Or even more explicit stub: `using DirectoryListerDelegate = nullptr_t;` This only defines a type, so that the public function of `DirectoryLister( .., DirectoryListerDelegate)` below gets correctly parsed/picked up.

Reply:
i see this error 

../..\net/base/directory_lister.h:66:25: error: unknown type name 'DirectoryListerData'; did you mean 'Directo
ryLister::DirectoryListerData'?
  void OnListFile(const DirectoryListerData& data);


seems like i need it